en/sql/tuning.rst

```diff
@@ -4189,3 +4191,178 @@ The user can check the query to be cached or not by putting the session command
+
+.. _correlated-subquery-cache:
+
+SUBQUERY CACHE (correlated)
+------------------------------------
+
+Subquery cache optimization can be used to enhance the performance of queries containing correlated subqueries, and the results of the subqueries are cached in independent spaces for each subquery.
+To disable subquery cache optimization, use the NO_SUBQUERY_CACHE hint on the target subquery.
+
+Subquery cache does not operate in the following scenarios:
+
+* When the correlated subquery contains another correlated subquery.
+* When the subquery is not in the SELECT clause.
+* When the subquery includes OID-related features.
+* When the subquery includes the NO_SUBQUERY_CACHE hint.
+* When storing new results exceeds the set subquery cache size (default: 2MB).
+* When the subquery contains functions that change results with each execution, such as random() or sys_guid().
+
+If the correlated subquery is in the SELECT clause, subquery cache is utilized.
+Among the recurrently executed correlated subqueries, if the column values referenced in the main query remain the same, the cached results are used to prevent re-execution.
+If no cached query is found, the subquery is processed and its results, along with the column values and query results, are cached.
+If the same column values are found in the cache, the results are retrieved from the cached area.
+
+The following example measures the performance difference depending on whether the subquery cache is used or not.
+First, a query to prepare the data for measuring performance differences is written: ::
+    
+    # Prepare data
+    csql> drop table if exists t1;
+    
+    csql> CREATE TABLE t1 AS
+            SELECT
+                    ROWNUM AS t1_pk,
+                    MOD(ROWNUM, 10) AS c1,
+                    MOD(ROWNUM, 100) AS c2,
+                    MOD(ROWNUM, 1000) AS c3
+            FROM 
+                    db_class a, db_class b, db_class c, db_class d, db_class e, db_class f
+            LIMIT 100000;
+    
+    csql> ALTER TABLE t1 ADD CONSTRAINT PRIMARY KEY pk_t1 (t1_pk);
+
+    csql> CREATE TABLE t2 AS
+            SELECT
+                    ROWNUM as c1,
+                    1 as c2,
+                    TO_CHAR(ROWNUM * 1000, '0999') as code
+            FROM 
+                    db_class a, db_class b
+            LIMIT 10;
+    
+    csql> update statistics on t1 with fullscan;
+    
+    csql> ;trace on  
+ 
+In CSQL, the improved performance can be easily measured by repeatedly executing queries using the COUNT function as shown in the example below.
+The results of the first subquery might be slow as the cache is not activated using the **NO_SUBQUERY_CACHE** hint, but from the second execution, it becomes much faster because it retrieves from the cached area: ::
```

Comment:
```suggestion
The results of the first example might be slow as the cache is not activated using the **NO_SUBQUERY_CACHE** hint, but from the second example, it becomes much faster because it retrieves from the cached area: ::
```